// @sanity/form-builder/lib/inputs/DateInputs/DateTimeInput.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.DateTimeInput = void 0;

var _react = _interopRequireDefault(require("react"));

var _CommonDateTimeInput = require("./CommonDateTimeInput");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

var DEFAULT_DATE_FORMAT = 'YYYY-MM-DD';
var DEFAULT_TIME_FORMAT = 'HH:mm';

function parseOptions() {
  var options = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : {};
  return {
    dateFormat: options.dateFormat || DEFAULT_DATE_FORMAT,
    timeFormat: options.timeFormat || DEFAULT_TIME_FORMAT,
    timeStep: 'timeStep' in options && Number(options.timeStep) || 1,
    calendarTodayLabel: options.calendarTodayLabel || 'Today'
  };
}

var DateTimeInput = /*#__PURE__*/_react.default.forwardRef(function DateTimeInput(props, forwardedRef) {
  var type = props.type,
      rest = _objectWithoutProperties(props, ["type"]);

  var title = type.title,
      description = type.description,
      placeholder = type.placeholder;

  var _parseOptions = parseOptions(type.options),
      dateFormat = _parseOptions.dateFormat,
      timeFormat = _parseOptions.timeFormat,
      timeStep = _parseOptions.timeStep;

  return /*#__PURE__*/_react.default.createElement(_CommonDateTimeInput.CommonDateTimeInput, _extends({}, rest, {
    ref: forwardedRef,
    selectTime: true,
    timeStep: timeStep,
    title: title,
    description: description,
    placeholder: placeholder,
    inputFormat: "".concat(dateFormat, " ").concat(timeFormat)
  }));
});

exports.DateTimeInput = DateTimeInput;
